receiver/awscontainerinsightreceiver/internal/neuron/neuron_empty_metric_decorator.go

```diff
@@ -0,0 +1,159 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package neuron
+
+import (
+	"context"
+	"strconv"
+	"strings"
+
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.uber.org/zap"
+)
+
+const (
+	statusType     = "status_type"
+	errorType      = "error_type"
+	memoryLocation = "memory_location"
+	percentile     = "percentile"
+)
+
+var attributeConfig = map[string][]string{
+	NeuronExecutionStatus:                       {statusType},
+	NeuronExecutionErrors:                       {errorType},
+	NeuronRuntimeMemoryUsage:                    {memoryLocation},
+	NeuronExecutionLatency:                      {percentile},
+	NeuronCoreUtilization:                       {neuronCoreAttributeKey, neuronDeviceAttributeKey},
+	NeuronCoreMemoryUtilizationConstants:        {neuronCoreAttributeKey, neuronDeviceAttributeKey},
+	NeuronCoreMemoryUtilizationModelCode:        {neuronCoreAttributeKey, neuronDeviceAttributeKey},
+	NeuronCoreMemoryUtilizationSharedScratchpad: {neuronCoreAttributeKey, neuronDeviceAttributeKey},
+	NeuronCoreMemoryUtilizationRuntimeMemory:    {neuronCoreAttributeKey, neuronDeviceAttributeKey},
+	NeuronCoreMemoryUtilizationTensors:          {neuronCoreAttributeKey, neuronDeviceAttributeKey},
+}
+
+var nonCoreAttributeValues = map[string]string{
+	statusType:     "completed",
+	errorType:      "generic",
+	memoryLocation: "neuron_device",
+	percentile:     "p50",
+}
+
+type EmptyMetricDecorator struct {
+	NextConsumer consumer.Metrics
+	Logger       *zap.Logger
+}
+
+func (ed *EmptyMetricDecorator) Capabilities() consumer.Capabilities {
+	return consumer.Capabilities{
+		MutatesData: true,
+	}
+}
+
+func (ed *EmptyMetricDecorator) ConsumeMetrics(ctx context.Context, md pmetric.Metrics) error {
+	rms := md.ResourceMetrics()
+	for i := 0; i < rms.Len(); i++ {
+		rs := rms.At(i)
+		ilms := rs.ScopeMetrics()
+		for j := 0; j < ilms.Len(); j++ {
+			ils := ilms.At(j)
+			metrics := ils.Metrics()
+
+			neuronHardwareInfo, neuronHardwareInfoFound := findNeuronHardwareInfo(metrics)
+			if neuronHardwareInfoFound {
+				ed.addEmptyMetrics(neuronHardwareInfo, metrics)
+			}
+		}
+	}
+	return ed.NextConsumer.ConsumeMetrics(ctx, md)
+}
+
+func (ed *EmptyMetricDecorator) addEmptyMetrics(hardwareInfo pmetric.Metric, metrics pmetric.MetricSlice) {
+	var metricFoundMap = make(map[string]bool)
+	for k := range attributeConfig {
+		metricFoundMap[k] = false
+	}
+
+	for k := 0; k < metrics.Len(); k++ {
+		m := metrics.At(k)
+		if _, ok := metricFoundMap[m.Name()]; ok {
+			metricFoundMap[m.Name()] = true
+		}
+	}
+
+	for k, v := range metricFoundMap {
+		if !v {
+			if strings.Contains(k, "core") {
+				populateCoreMetrics(metrics, k, attributeConfig[k], hardwareInfo)
```

Comment:
To help my understanding, how do you ensure that only node-level metrics (not container or pod) are populated?  Is it because you don't have the PodName/ContainerName attributes?

Reply:
Yes correct, and we replicate the node level metrics to pods, plus if the workload is not running we dont have pod correlation happening